sensor/kubernetes/crs/crs.go

```diff
@@ -0,0 +1,299 @@
+func temporarilyStoreRegistrantSecret(crs *crs.CRS) error {
+	// Extract (first) CA certificate from the CRS.
+	var caCert string
+	if len(crs.CAs) > 0 {
+		caCert = crs.CAs[0]
+	}
+	if caCert == "" {
+		return errors.New("malformed Cluster Registration Secret (missing CA certificate)")
+	}
+
+	for fileName, content := range map[string]string{
+		"ca.pem":   caCert,
+		"cert.pem": crs.Cert,
+		"key.pem":  crs.Key,
+	} {
+		filePath := filepath.Join(mtls.CertsPrefix, fileName)
+		if err := os.WriteFile(filePath, []byte(content), 0600); err != nil {
+			return errors.Wrapf(err, "writing mTLS material to file %q", filePath)
+		}
+		log.Infof("Successfully wrote file %s.", filePath)
+	}
+
+	return nil
+}
+
+func openCentralConnection() (*grpcUtil.LazyClientConn, error) {
+	// Create central client.
+	centralEndpoint := env.CentralEndpoint.Setting()
+	centralClient, err := centralclient.NewClient(centralEndpoint)
+	if err != nil {
+		return nil, errors.Wrapf(err, "initializing Central client for endpoint %s", env.CentralEndpoint.Setting())
+	}
+
+	centralConnFactory := centralclient.NewCentralConnectionFactory(centralClient)
+	centralConnection := grpcUtil.NewLazyClientConn()
+	certLoader := centralclient.RemoteCertLoader(centralClient)
+	go centralConnFactory.SetCentralConnectionWithRetries(centralConnection, certLoader)
+
+	log.Infof("Connecting to Central server %s", centralEndpoint)
+
+	// Wait for connection to be ready.
+	okSig := centralConnFactory.OkSignal()
+	errSig := centralConnFactory.StopSignal()
+	select {
+	case <-errSig.Done():
+		return nil, errors.Wrap(errSig.Err(), "waiting for Central connection from factory")
+	case <-okSig.Done():
+	}
+
+	log.Info("Central connection ready.")
+	return centralConnection, nil
+}
+
+// centralHandshake performs the hello-handshake with Central and returns Central's CentralHello reponse on success.
+func centralHandshake(ctx context.Context, k8sClient kubernetes.Interface, centralConnection *grpcUtil.LazyClientConn) (*central.CentralHello, error) {
+	sensorHello, err := prepareSensorHelloMessage(k8sClient)
+	if err != nil {
+		return nil, errors.Wrap(err, "preparing SensorHello message")
+	}
+
+	ctx = metadata.AppendToOutgoingContext(ctx, centralsensor.SensorHelloMetadataKey, "true")
+	ctx, err = centralsensor.AppendSensorHelloInfoToOutgoingMetadata(ctx, sensorHello)
+	if err != nil {
+		return nil, errors.Wrap(err, "appending SensorHello info to outgoing metadata")
+	}
+
+	client := central.NewSensorServiceClient(centralConnection)
+	stream, err := sensorCommon.CommunicateWithAutoSensedEncoding(ctx, client)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating bidirectional stream with auto-sensed encoding")
+	}
+
+	rawHdr, err := stream.Header()
+	if err != nil {
+		return nil, errors.Wrap(err, "receiving headers from Central")
+	}
+
+	hdr := metautils.MD(rawHdr)
+	if hdr.Get(centralsensor.SensorHelloMetadataKey) != "true" {
```

Comment:
ditto

Reply:
Would prefer to not do this as part of this PR.